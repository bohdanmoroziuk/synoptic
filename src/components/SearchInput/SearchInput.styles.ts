import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    backgroundColor: '#666',
    color: 'white',
    height: 40,
    width: '100%',
    marginTop: 20,
    marginHorizontal: 40,
    paddingHorizontal: 10,
    alignSelf: 'center',
    borderRadius: 5,
  },
  textInput: {
    flex: 1,
    color: 'white',
  },
});
