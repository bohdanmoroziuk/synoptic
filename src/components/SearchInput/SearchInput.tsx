import { FunctionComponent } from 'react';
import { View, TextInput, TextInputProps } from 'react-native';

import styles from './SearchInput.styles';

export interface SearchInputProps {
  placeholder: TextInputProps['placeholder'];
}

const SearchInput: FunctionComponent<SearchInputProps> = ({ placeholder }) => {
  return (
    <View style={styles.container}>
      <TextInput
        style={styles.textInput}
        autoCorrect={false}
        placeholder={placeholder}
        placeholderTextColor="white"
        underlineColorAndroid="transparent"
        clearButtonMode="always"
      />
    </View>
  );
};

export default SearchInput;
