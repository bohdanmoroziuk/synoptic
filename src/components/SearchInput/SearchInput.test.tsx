import { render } from '@testing-library/react-native';

import SearchInput from './SearchInput';

describe('SearchInput', () => {
  it('renders successfully', () => {
    const mockPlaceholder = 'Type here';

    const { toJSON } = render(<SearchInput placeholder={mockPlaceholder} />);

    expect(toJSON()).toMatchSnapshot();
  });
});
