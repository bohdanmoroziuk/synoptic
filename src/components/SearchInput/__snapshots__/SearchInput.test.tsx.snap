// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput renders successfully 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "#666",
      "borderRadius": 5,
      "color": "white",
      "height": 40,
      "marginHorizontal": 40,
      "marginTop": 20,
      "paddingHorizontal": 10,
      "width": 300,
    }
  }
>
  <TextInput
    autoCorrect={false}
    clearButtonMode="always"
    placeholder="Type here"
    placeholderTextColor="white"
    style={
      Object {
        "color": "white",
        "flex": 1,
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
